### ISSUE TITLE
Stackups where the first entry in min spacing isn't exactly equal to pitch - min width cause issues

### ISSUE THREAD

**Issue by edwardcwang** [ucb-bar/hammer#429]
Sometimes the stackup information is emitting a first entry where `min_spacing > pitch - min_width`. e.g.  if pitch=0.20, min_width=0.05, and the first entry has min_spacing of 0.01. This causes an unwanted assertion error which blocks running the design:

`AssertionError: Tech plugin is malformed for metal M2, the minimum spacing in the width-spacing list must be the same as (pitch - min_width)`

The short-term hack/fix is to change the first tuple's min_spacing to match `pitch - min_width` at the risk of possible impact to QoR.

The longer term fix is to use correct spacing values, though it is currently a bit unclear as to how to do that.

**Comment by jwright6323** [comment]
To add to the possible workarounds- you could always roll your own power straps hook (which was required before the API existed anyway)

**Comment by jwright6323** [comment]
Is this still high priority?

**Comment by edwardcwang** [comment]
Would be great to not need to hack core hammer or the tech plugin in cases like this as it may impede reuse...

**Comment by colinschmidt** [comment]
Agreed. But high priority implies someone is blocked or working on it. So I think the new classification is reasonable.

**Comment by harrisonliew** [comment]
We ran into this for ASAP7. Conveniently, a spacing table isn't defined for the affected layers so we put it manually in the tech json (this is for documentation purposes)
